libs/dal/src/repositories/message/message.repository.ts

```diff
@@ -260,15 +308,24 @@ export class MessageRepository extends BaseRepository<EnforceEnvironmentQuery, M
     );
   }
 
-  async delete(query: EnforceEnvironmentQuery) {
-    const message = await this.findOne({ _id: query._id, _environmentId: query._environmentId });
+  @InvalidateCache()
+  async delete(query: EnforceSubscriberQuery) {
+    const message = await this.findOne({
+      _id: query._id,
+      _environmentId: query._environmentId,
+    } as EnforceSubscriberQuery);
     if (!message) {
       throw new DalException(`Could not find a message with id ${query._id}`);
     }
+
+    if (isStoreConnected(this.cacheService?.getStatus())) {
```

Comment:
```suggestion
    if (this.cacheService?.cacheEnabled()) {
```

Can we just do something like this instead?